File: python/queue.py

class Queue:
    def __init__(self, reverse=False):
        self.items = []
        self.reverse = reverse

    @property
    def length(self):
        return len(self.items)

    @property
    def empty(self):
        if self.length:
            return True
        else:
            return False

    def enqueue(self, x):
        self.items = self.items + [x]

    def dequeue(self):
        self.items = self.items[1:]

    def import_list(self, lst):
        if not self.reverse:
            for i in sorted(lst):
                self.enqueue(i)
        else:
            for i in sorted(lst)[::-1]:
                self.enqueue(i)

File: java/Queue.java
import java.util.ArrayList;

public class Queue {
    private ArrayList items;

    public Queue() {
        items = new ArrayList<>();
    }

    public int length() {
        return items.size();
    }

    public boolean empty() {
        return items.isEmpty();
    }

    public void enqueue(Object x) {
        items.add(x);
    }

    public void dequeue() {
        items.remove(0);
    }
}